app/src/main/java/tk/therealsuji/vtopchennai/fragments/RecyclerViewFragment.java

```
@@ -61,6 +64,21 @@ public void onSuccess(@NonNull List<Receipt> receipts) {
                             return;
                         }
 
+// Receipts are sorted by date in descending order(newest first) - Code Starts Here
+                        Collections.sort(receipts, new Comparator<Receipt>() {
+                            @Override
+                            public int compare(Receipt r1, Receipt r2) {
+                                @SuppressLint("SimpleDateFormat") SimpleDateFormat format = new SimpleDateFormat("dd-MMM-yyyy");
+                                try {
+                                    return format.parse(r2.getDate()).compareTo(format.parse(r1.getDate()));
+                                } catch (Exception e) {
+                                    e.printStackTrace();
+                                }
+                                return 0;
+                            }
+                        });
+// Code Ends Here
```

A few changes here,

- The comments are unnecessary, it's understood what's going on here.
- Avoid suppressing warnings (`@SuppressLint`). The warning appears due to the missing locale.
  ```java
  SimpleDateFormat format = new SimpleDateFormat("dd-MMM-yyyy", Locale.ENGLISH);
  ```
- Printing the stack trace is unnecessary, ignore the exception instead.
  ```diff
  - } catch (Exception e) {
  + } catch (Exception ignored) {
  -     e.printStackTrace();
    }
  ```